kotest-extensions/src/jvmMain/kotlin/io/kotest/extensions/system/SystemPropertiesExtensions.kt

```diff
@@ -192,6 +193,30 @@ class SystemPropertyTestListener(newProperties: Map<String, String?>, mode: Over
    override suspend fun afterTest(testCase: TestCase, result: TestResult) {
       resetSystemProperties()
    }
+
+   override suspend fun beforeContainer(testCase: TestCase) {
+      if (testCase.type == TestType.Container) changeSystemProperties()
+   }
+
+   override suspend fun afterContainer(testCase: TestCase, result: TestResult) {
+      if (testCase.type == TestType.Container) resetSystemProperties()
+   }
+
+   override suspend fun beforeEach(testCase: TestCase) {
+      if (testCase.type == TestType.Test) changeSystemProperties()
+   }
+
+   override suspend fun afterEach(testCase: TestCase, result: TestResult) {
+      if (testCase.type == TestType.Test) resetSystemProperties()
+   }
+
+   override suspend fun beforeAny(testCase: TestCase) {
+      changeSystemProperties()
+   }
+
+   override suspend fun afterAny(testCase: TestCase, result: TestResult) {
+      resetSystemProperties()
```

Comment:
We probably only need to call changeSystemProperties resetSystemProperties on one of the beforeAny / beforeTest cases as that will cover it for every test.